modular_skyrat/modules/modular_items/lewd_items/code/lewd_helpers/human.dm

```diff
@@ -0,0 +1,450 @@
+/mob/living/carbon/human
+	var/arousal = 0
+	var/pleasure = 0
+	var/pain = 0
+
+	var/pain_limit = 0
+	var/arousal_status = AROUSAL_NONE
+
+	// Add variables for slots to the human class
+	var/obj/item/vagina = null
+	var/obj/item/anus = null
+	var/obj/item/nipples = null
+	var/obj/item/penis = null
+
+	var/has_penis = FALSE
+	var/has_vagina = FALSE
+	var/has_breasts = FALSE
+	var/has_anus = FALSE
+
+
+// For tracking arousal and fluid regen.
+/mob/living/carbon/human/Initialize(mapload)
+	. = ..()
+	if(!istype(src, /mob/living/carbon/human/species/monkey))
+		apply_status_effect(/datum/status_effect/aroused)
+		apply_status_effect(/datum/status_effect/body_fluid_regen)
+
+/*
+*	This code needed to determine if the human is naked in that part of body or not
+*	You can use this for your own stuff if you want, haha.
+*/
+
+/// Are we wearing something that covers our chest?
+/mob/living/carbon/human/proc/is_topless()
+	return (!(wear_suit) || !(wear_suit.body_parts_covered & CHEST)) && (!(w_uniform) || !(w_uniform.body_parts_covered & CHEST))
+
+/// Are we wearing something that covers our groin?
+/mob/living/carbon/human/proc/is_bottomless()
+	return (!(wear_suit) || !(wear_suit.body_parts_covered & GROIN)) && (!(w_uniform) || !(w_uniform.body_parts_covered & GROIN))
+
+/// Are we wearing something that covers our shoes?
+/mob/living/carbon/human/proc/is_barefoot()
+	return (!(wear_suit) || !(wear_suit.body_parts_covered & GROIN)) && (!(shoes) || !(shoes.body_parts_covered & FEET))
+
+/mob/living/carbon/human/proc/is_hands_uncovered()
+    return (gloves?.body_parts_covered & ARMS)
+
+/mob/living/carbon/human/proc/is_head_uncovered()
+    return (head?.body_parts_covered & HEAD)
+
+/mob/living/carbon/human/proc/has_penis(required_state = REQUIRE_GENITAL_ANY)
+	if(issilicon(src) && has_penis)
+		return TRUE
+	var/obj/item/organ/external/genital/peepee = getorganslot(ORGAN_SLOT_PENIS)
+	if(peepee)
+		switch(required_state)
+			if(REQUIRE_GENITAL_ANY)
+				return TRUE
+			if(REQUIRE_GENITAL_EXPOSED)
+				if(peepee.visibility_preference == GENITAL_ALWAYS_SHOW || is_bottomless())
+					return TRUE
+				else
+					return FALSE
+			if(REQUIRE_GENITAL_UNEXPOSED)
+				if(peepee.visibility_preference != GENITAL_ALWAYS_SHOW && !is_bottomless())
+					return TRUE
+				else
+					return FALSE
```

Comment:
This should be on a singular line, rather than being an if-else with two return statements.